### PR TITLE
Added DNS Reverse Record logic for service IPs

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

File: pkg/dns/dns.go [hunk 1]
```diff
@@ -284,9 +291,13 @@ func (kd *KubeDNS) newPortalService(service *kapi.Service) {
 		}
 	}
 	subCachePath := append(kd.domainPath, serviceSubdomain, service.Namespace)
+	host := kd.getServiceFQDN(service)
+	reverseRecord, _ := getSkyMsg(host, 0)
+
 	kd.cacheLock.Lock()
 	defer kd.cacheLock.Unlock()
 	kd.cache.setSubCache(service.Name, subCache, subCachePath...)
+	kd.reverseRecordMap[service.Spec.ClusterIP] = reverseRecord
```
Comment (thockin) [inline pkg/dns/dns.go]: should we add reverse records for petset pods (with hostname and subdomain set AND a governing headless service?  Maybe that's yet another followup.  Can you just file an issue on that?

**Comment by thockin** [comment]
LGTM

please build and push the image and get green on tests.

**Comment by thockin** [comment]
status on this?  It's marked "do not merge"...

**Comment by ArtfulCoder** [comment]
will rebase today and track the merge.

**Comment by ArtfulCoder** [comment]
I marked it as do-not-merge since I wanted the dnsmasq PR to go in first.
I will take care of this today.

**Comment by thockin** [comment]
dnsmasq is in

**Comment by ArtfulCoder** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by ArtfulCoder** [comment]
blocked by https://github.com/kubernetes/kubernetes/pull/26335

**Comment by ArtfulCoder** [comment]
the test for the PR fails because #26335 is not yet merged.

**Comment by wojtek-t** [comment]
I think test failure may be related...

**Comment by ArtfulCoder** [comment]
rebased. waiting for test run.

**Comment by ArtfulCoder** [comment]
@girishkalele can you look into this.
The failure is  related to the PR change.
I rebaesd yest. in the hope that the test would pass but it didnt :(

**Comment by ArtfulCoder** [comment]
@girishkalele you might want to do a new kube-dns image with the code of the PR and then try.. hopefully, it is a simple case of pushing a new kube-dns image.

**Comment by girishkalele** [comment]
@ArtfulCoder 

If I understand correctly, I checkout your PR, build a new kube-dns from your code and push to Google Containers. Will do.

**Comment by girishkalele** [comment]
@k8s-bot test this issue: #IGNORE

File: pkg/dns/dns.go (hunk 1 above)
Comment (girishkalele) [inline pkg/dns/dns.go]: Filed issue https://github.com/kubernetes/kubernetes/issues/26752 to track this.

**Pull request merged**